kinto/core/resource/__init__.py

```diff
@@ -622,6 +634,18 @@ def apply_changes(self, record, changes):
         :returns: the new record with `changes` applied.
         :rtype: dict
         """
+        content_type = str(self.request.headers.get('Content-Type'))
+
+        if content_type == 'application/json-patch+json':
+            try:
+                changes, _ = apply_json_patch(record, changes)
+            except (JsonPatchException, JsonPointerException) as e:
+                error_details = {
+                    'name': '',
+                    'description': 'JSON Patch operation failed'
+                }
+                raise_invalid(self.request, e, **error_details)
+
```

Comment:
here I would do a `else`/`else if` for the rest of this method, since what's coming below does not apply to json patch. 

plus, it would reduce the complexity for the reader, where each block applies to a particular case